sympy/matrices/dense.py

```diff
@@ -881,7 +881,10 @@ def row_del(self, i):
         col_del
         """
         if i < -self.rows or i >= self.rows:
-            raise IndexError("Index out of range: 'i = %s', valid -%s <= i < %s" % (i, self.rows, self.rows))
+            raise IndexError("Index out of range: 'i = %s', valid -%s <= i"
+                             " < %s" % (i, self.rows, self.rows))
+        if i < 0:
+            i = self.rows - abs(i)
```

Comment:
That is cleaner. I will update shortly.
On 31 Aug 2015 20:52, "Kalevi Suominen" <notifications@github.com> wrote:

> In sympy/matrices/dense.py
> <https://github.com/sympy/sympy/pull/9880#discussion_r38322712>:
>
> > @@ -881,7 +881,10 @@ def row_del(self, i):
> >          col_del
> >          """
> >          if i < -self.rows or i >= self.rows:
> > -            raise IndexError("Index out of range: 'i = %s', valid -%s <= i < %s" % (i, self.rows, self.rows))
> > +            raise IndexError("Index out of range: 'i = %s', valid -%s <= i"
> > +                             " < %s" % (i, self.rows, self.rows))
> > +        if i < 0:
> > +            i = self.rows - abs(i)
>
> Or preferably i += self.rows.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sympy/sympy/pull/9880/files#r38322712>.
>